starter/src/main/java/org/teiid/spring/autoconfigure/TeiidAutoConfiguration.java

```diff
@@ -239,4 +254,18 @@ public PhysicalNamingStrategy teiidNamingStrategy() {
             return null;
         }
     }
-}
+
+    @Bean
+    @Primary
+    public PlatformTransactionManager platformTransactionManager() {
+        if (this.transactionManager != null) {
+            return new JtaTransactionManager(this.userTransaction, this.transactionManager);
+        }
+        return new DelegatingPlatformTransactionManager();
```

Comment:
Sorry should have been more specific. I meant in the check method.

Looking at this again, I'm still not sure that we should be taking responsibility for the construction of the platform transaction manager.  It seems like that should already be part of the application - or an optional auto configuration - not something that is primary by default.

Reply:
The issue is the default one does not do the chained data sources, only for the single data source. I believe, if we do that, that single data source is Teiid's data source, then Teiid has to manage the underlying data sources it is using. We can probably set the chained txn manager only on Teiid, and let Teiid take care of the commit cycle based on the application level platform transaction manager.

Reply:
@shawkins Taking look at Spring code again, the default PlatformTransactionManager is *only* created when there is single "dataSource" is created. When there are two or more it simply does not create any and leaves it up to user looks like as I see NULL in that case for PTM. In Teiid case, it is always going to be at least 2 data sources in the config, one Teiid itself and then any "source" data source. So, with the above code, Teiid will either create a default PTM as Spring does ignoring the Teiid data source, and handles the multiple data source case. WDYT?

Reply:
That is what I mean by adding some magic for case 1 - the single data source case which would have been handled implicitly without Teiid.  My concern is that in the other 2 cases the user is already responsible for the setup of the platform transaction manager and we shouldn't cause a conflict there.

Reply:
Took your advice, and built the logic into `MultiDataSourceTransactionManagement` class that a DataSource class can extend to bring in the functionality. Which will keep the control with the developer but when he/she extends DataSources class with `MultiDataSourceTransactionManagement` it will bring the multi-source platform transaction manager into scope.